frontend/test/pytest/test_cuda_integration.py

```diff
@@ -502,5 +503,173 @@ def circuit2(a):
         assert_allclose(expected, observed)
 
 
+@pytest.mark.cuda
+class TestForLoops:
+    """Test for loops."""
+
+    def test_basic_loop(self):
+        """Test simple for loop."""
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_lightning(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_):
+                qml.PauliX(0)
+
+            loop_fn()
+            return qml.state()
+
+        from catalyst.cuda import SoftwareQQPP
+
+        @qml.qnode(SoftwareQQPP(wires=1))
+        def circuit(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_):
+                qml.PauliX(0)
+
+            loop_fn()
+            return qml.state()
```

Comment:
Maybe you could reformulate the test so it doesn't depend on comparing against a second execution? :) For very simple cases like this it should be possible.

Reply:
Done, the new test cases does not depend on the lightning backend.